src/other/Comms.js

```diff
@@ -927,7 +927,7 @@
         function filterResults(request) {
             var changedFilter = {};
             for (var key in request) {
-                if (request[key] !== undefined && request[key + "_changed"] !== undefined) {
+                if (request[key + "_changed"] !== undefined) {
                     changedFilter[key] = {
                         value: request[key]
```

Comment:
you are looping through the request on the keys so I assume request[key] could never be a NPE... but I'm not sure what you are gaining by removing request[key] !== undefined since that should never be the case should it?  Would it also not be the case that the for (var key in request) would also contain the key that you are building from request[key +"_changed"]... maybe I'm missing something.